gbfs.md

```diff
@@ -269,28 +245,34 @@ Example:
+\- num_docks_disabled  | Optional  | Number of empty but disabled dock points at the station. This value remains as part of the spec as it is possibly useful during development
 \- is_installed        | Yes       | 1/0 boolean - is the station currently on the street
-\- is_renting          | Yes       | 1/0 boolean - is the station currently renting vehicles (even if the station is empty, if it is set to allow rentals this value should be 1)
-\- is_returning        | Yes       | 1/0 boolean - is the station accepting vehicle returns (if a station is full but would allow a return if it was not full then this value should be 1)
+\- is_renting          | Yes       | 1/0 boolean - is the station currently renting bikes (even if the station is empty, if it is set to allow rentals this value should be 1)
+\- is_returning        | Yes       | 1/0 boolean - is the station accepting bike returns (if a station is full but would allow a return if it was not full then this value should be 1)
 \- last_reported       | Yes       | Integer POSIX timestamp indicating the last time this station reported its status to the backend
-\- num_vehicles_available | Yes       | An object consisting of keys that are vehicle_type_ids and values that represent the number of vehicles of the respective vehicle type available for rental
-\- num_vehicles_disabled  | Optional  | An object consisting of keys that are vehicle_type_ids and values that represent the number of disabled vehicles of the respective vehicle type at the station. Vendors who do not want to publicize the number of disabled vehicles or docks in their system can opt to omit station capacity (in station_information), num_vehicles_disabled and num_docks_disabled. If station capacity is published then broken docks/vehicles can be inferred (though not specifically whether the decreased capacity is a broken vehicle or dock)
-\- num_docks_available | Yes       | An object consisting of keys that are vehicle_type_ids and values that represent the number of docks accepting vehicle returns of vehicles of the respective vehicle type
-\- num_docks_disabled  | Optional  | An object consisting of keys that are vehicle_type_ids and values that represent the number of empty but disabled dock points for vehicles of the respective vehicle type at the station. This value remains as part of the spec as it is possibly useful during development
+\- vehicle_docks_available | Conditionally Required       | This field is required if the [vehicle_types.json](#vehicle_typesjson) file has been defined and the `num_docks_available` field is not defined. This field's value is an object consisting of keys that are vehicle_type_ids as described in [vehicle_types.json](#vehicle_typesjson) and values that represent the number of docks accepting vehicle returns of vehicles of the respective vehicle type. If a single dock can accept multiple types, these should be added to the count of available docks for each applicable vehicle type.
```

Comment:
It seems like this change makes it difficult for clients to understand how many total docks are available.  For example, let's say the station accepts bikes, e-bikes, and scooters.  There are 5 bike docks, 5 e-bike docks, and 5 scooter docks available.  So are there 5 available docks if you have a bike or e-bike, or 10?  How would you know that the docks are specific in what they can accept or not?

Additionally, this requires knowledge from another feed, which makes it especially difficult to understand the dock situation.

Reply:
If the optional `station_information`.`capacity` field was provided, then consumers could know the total capacity. I think  appropriate spec here depends on the needs of consumers. If consumers need more precision, then the proposed `vehicle_docks_available` object could instead include an array of `vehicle_type_id`.

What do you think?
@kanagy @gcamp

Reply:
I've also been wondering this and do believe it would be challenging to make a UI that shows dock availability according to vehicle type. Overall it does seem like it would be useful to provide an overall metric of the total number of docks available, but I still think it's critical to include vehicle-specific dock availability info as well. 

Looking at how I worded the spec, I think it would make sense to make a change to always require the `station_information.json#capacity` and `station_status.json#num_docks_available` fields and indicate that it always represents the total amount of docks regardless of whether certain docks only accept certain vehicle types. In a system that omits the vehicle_types.json file, this would indicate the number of docks at the station (according to `station_information.json#capacity`) and those that are currently available (according to `station_status.json#num_docks_available`). And in systems that include the vehicle_types.json this could represent the total number of docks/current availability of docks regardless of the compatibility of each dock with each vehicle type.

Reply:
I agree with making `station_status.num_docks_available` always required.  In addition, you may want to change `vehicle_docks_available` so you can represent exclusivity better.  For example: `vehicle_docks_available: [{'vehicle_type': ['ebike', 'bike'], 'count': 5}]`.